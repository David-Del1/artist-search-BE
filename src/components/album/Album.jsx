import React, { useEffect, useState } from 'react';
import { fetchCoverArt } from '../../services/coverArtApi';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function Album({ title, id }) {
  const [url, setUrl] = useState('');

  useEffect(() => {
    (async () => {
      const link = await fetchCoverArt(id);
      setUrl(link);
    })();
  });

  return (
    <section>
      <Link to={`/artist/album/${id}`}>
        <h1>{title}</h1>
        <img src={url} alt={title} />
      </ Link>
    </section>
  );
}

Album.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired
};

export default Album;

