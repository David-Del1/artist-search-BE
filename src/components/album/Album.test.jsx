import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Album from './Album';

describe('album component', () => {
  it('makes a album', () => {
    const { asFragment } = render(

      <MemoryRouter>
        <Album />
      </MemoryRouter>

    );

    expect(asFragment()).toMatchSnapshot();
  });
});
