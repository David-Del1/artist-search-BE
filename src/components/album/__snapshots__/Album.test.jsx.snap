// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album component makes a album 1`] = `
<DocumentFragment>
  <section>
    <a
      href="/artist/album/undefined"
    >
      <h1 />
      <img
        src=""
      />
    </a>
  </section>
</DocumentFragment>
`;
